Lib/fontTools/ttLib/tables/_k_e_r_n.py

```diff
@@ -33,21 +33,23 @@ def decompile(self, data, ttFont):
 		else:
 			self.version = version
 			data = data[4:]
-		tablesIndex = []
 		self.kernTables = []
 		for i in range(nTables):
 			if self.version == 1.0:
 				# Apple
-				length, coverage, tupleIndex = struct.unpack(">lHH", data[:8])
-				version = coverage & 0xff
+				length, coverage = struct.unpack(">LH", data[:6])
 			else:
-				version, length = struct.unpack(">HH", data[:4])
-			length = int(length)
-			if version not in kern_classes:
-				subtable = KernTable_format_unkown(version)
+				# in OpenType spec the "version" field refers to the common
+				# subtable header; the actual subtable format is stored in
+				# the last 8 mask bits of "coverage" field.
+				# Since this "version" is always 0 (and is not present in the
+				# later AAT extensions), we simply ignore it here
+				_, length, coverage = struct.unpack(">HHH", data[:6])
+			subtableFormat = coverage & 0xff
```

Comment:
ooh.... actually also the order of format and coverage bytes is different between OT and AAT kern! In OT the first byte is format and the following one is coverage, whereas in AAT it's the reverse. I got this completely wrong, sorry.. :grimacing: 
Thank @justvanrossum that had me check that CALIBRI.TTF kern. I was surprised when I say a microsoft font with a kern subtable format _1_ (Apple-only). It turns out I was reading wrong, the 1 was the coverage bit (horizontal data), not the format.
Why did they make it so complicated :confounded:

Reply:
Yeah it's as if someone took the existing format and said let's change everything about it!  Allocate higher byte for coverage, allocate bits MSB-to-LSB, and allocate a bit for horizontal, instead of vertical :)).